share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml

```diff
@@ -1,7 +1,7 @@
 stages: [ "generate", "build" ]
 
 default:
-  image: { "name": "ghcr.io/scottwittenburg/ecpe4s-ubuntu18.04-runner-x86_64:2020-09-01", "entrypoint": [""] }
+  image: { "name": "ghcr.io/scottwittenburg/ecpe4s-ubuntu18.04-runner-x86_64:2021-05-15", "entrypoint": [""] }
```

Comment:
Did you already check that the compilers this image has are the same as the `2020-09-01` tagged image?  If not, pipeline jobs will end up doing a lot of duplicated bootstrapping.  So in that case, we would either need to change the compiler we specify in the stacks that use the default image (at least `share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml`, but also maybe others), or else make a new image with the desired compilers already installed.

Reply:
No, I didn't. For build-systems it might not matter but I think it matters for e4s. I'll stop the pipeline.

Reply:
@tgamblin Would it be fine with you if in this PR I add logic to use the original concretizer for the e4s pipeline, until #25287 is ready to be merged?

Reply:
Phew, that explains why it was canceled!  😅  I was worried there for a sec, when I didn't see any new push or request for spackbot to start a pipeline.

Reply:
So, after a chat on slack with @eugeneswalker it seems to me that we'll be using the same compilers as before, so I'll restart the pipeline and see how many rebuilds it triggers.